File: bob/bob.lua
local function is_question(phrase)
  return phrase:sub(#phrase,#phrase) == '?'
end

local function is_yelling(phrase)
  for i=1,#phrase do
    local c = phrase:sub(i,i)

    if c ~= string.upper(c) then
      return false
    end
  end
  return true
end

local function hey(phrase)
  if phrase == '' then return 'Fine, be that way.' end

  if is_question(phrase) then return 'Sure' end

  if is_yelling(phrase) then return 'Whoa, chill out!' end

  return 'Whatever'
end

return {
  hey = hey
}

File: bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      String.ends_with?(input, "?") ->
        "Sure."

      String.match?(input, ~r/^\s*$/) ->
        "Fine. Be that way!"

      input == String.upcase(input) and input != String.downcase(input) ->
        "Whoa, chill out!"

      true ->
        "Whatever."
    end
  end
end
